src/main/java/net/minecraftforge/common/conditions/ConditionHelper.java

```diff
@@ -27,135 +23,121 @@
-	{
-	    ResourceLocation key = serializer.getID();
-	    if (conditions.containsKey(key))
-	        throw new IllegalStateException("Duplicate recipe condition serializer: " + key);
-	    conditions.put(key, serializer);
-	    return serializer;
-	}
+    private static final Logger LOGGER = LogUtils.getLogger();
 
-	/**
-	 * Serializes a single condition as a {@link JsonObject}
-	 * @param condition The condition to be serialized
-	 * @return The serialized JSON
-	 * @throws JsonSyntaxException If no serializer is registered for the condition
-	 */
-	@SuppressWarnings({ "rawtypes", "unchecked" })
-	public static JsonObject serialize(ICondition condition)
-	{
-	    IConditionSerializer serializer = conditions.get(condition.getSerializerId());
-	    if (serializer == null)
-	    {
-	        throw new JsonSyntaxException("Unknown condition type: " + condition.getSerializerId().toString());
-	    }
-	    return serializer.getJson(condition);
-	}
+    /**
+     * Serializes a single condition as a {@link JsonObject}
+     * @param condition The condition to be serialized
+     * @return The serialized JSON
+     * @throws RuntimeException If no serializer is registered for the condition, or if the serializer encountered an exception encoding
+     */
+    public static JsonElement serialize(ICondition condition)
+    {
+        return ICondition.DIRECT_CODEC.encodeStart(JsonOps.INSTANCE, condition)
+                .getOrThrow(false, msg -> LOGGER.error("Encountered exception encoding condition: {}", msg));
+    }
 
     /**
      * Serializes an array of conditions as a {@link JsonArray}
      * @param conditions The conditions to be serialized
      * @return The serialized JSON
-     * @throws JsonSyntaxException If no serializer is registered for any of the passed conditions.
+     * @throws RuntimeException If no serializer is registered for any of the passed conditions.
      */
-	public static JsonArray serialize(ICondition... conditions)
-	{
-	    JsonArray arr = new JsonArray();
-	    for(ICondition iCond : conditions)
-	    {
-	        arr.add(ConditionHelper.serialize(iCond));
-	    }
-	    return arr;
-	}
-
-	/**
-	 * Parses an {@link ICondition} from a {@link JsonObject}.
-	 * @param json The serialized condition JSON
-	 * @return The deserialized condition object
-	 * @throws JsonSyntaxException If no serializer is registered for the specified condition type
-	 */
-	public static ICondition getCondition(JsonObject json)
-	{
-	    ResourceLocation type = new ResourceLocation(GsonHelper.getAsString(json, "type"));
-	    IConditionSerializer<?> serializer = ConditionHelper.conditions.get(type);
-	    if (serializer == null)
-	        throw new JsonSyntaxException("Unknown condition type: " + type.toString());
-	    return serializer.read(json);
-	}
+    public static JsonArray serialize(ICondition... conditions)
+    {
+        JsonArray arr = new JsonArray();
+        for(ICondition iCond : conditions)
```

Comment:
Formatting: space between `for` and opening parenthesis.
```suggestion
        for (ICondition iCond : conditions)
```